File: C++/PAT(Basic_Level)Practice copy/1002.cpp
/*
1.char转int：char-'0'
2.int转string：to_string(int)
3.string转int：stoi(string)
*/
#include<iostream>
#include<string>
#include<vector>
using namespace std;

int main() {
    vector<string> list = { "ling", "yi", "er", "san", "si", "wu", "liu", "qi", "ba", "jiu" };
    string s;
    cin >> s;
    int32_t sum = 0;
    for (auto& i : s) {
        sum += i - '0';
    }
    string res = to_string(sum);
    string s2 = "";
    for (auto& i : res) {
        int32_t a = i - '0';
        if (s2 == "") {
            s2 += list[a];
        }
        else {
            s2 += " ";
            s2 += list[a];
        }
    }
    cout << s2 << endl;

}

File: C#/PAT_B/1002.cs
using System;
class p_1002
{
    static string[] list = { "ling", "yi", "er", "san", "si", "wu", "liu", "qi", "ba", "jiu" };
    static void p1002()
    {
        var s = Console.ReadLine();
        if (s != "")
        {
            var sum = 0;
            foreach (var i in s!)
            {
                sum += i - '0';
            }
            var res = sum.ToString();
            for (var i = 0; i < res.Length; i++)
            {
                if (i != res.Length - 1)
                {
                    Console.Write("{0} ", list[res[i] - '0']);
                }
                else
                {
                    Console.WriteLine("{0}", list[res[i] - '0']);
                }
            }
        }
    }
}